test/specs/core.spec.js

```diff
@@ -259,6 +259,36 @@ describe('Riot core api', () => {
       riot.unregister('my-component')
     })
 
+    it('custom components can be unmounted by tag keeping the root element', () => {
+      const destroyedSpy = spy()
+      riot.register('my-component-red', {
+        css: 'my-component-red { color: red; }',
+        exports: {
+          onUnmounted() {
+            destroyedSpy()
+          }
+        }
+      })
+      riot.register('my-component-green', {
+        css: 'my-component-green { color: green; }',
+        exports: {
+          onUnmounted() {
+            destroyedSpy()
+          }
+        }
+      })
+      const element = document.createElement('div')
+
+      riot.mount(element, {}, 'my-component-red')
+      riot.unmount(element, true)
+      riot.mount(element, {}, 'my-component-green')
+      riot.unmount(element, true)
+
+      expect(destroyedSpy).to.have.been.calledTwice
```

Comment:
Could you please check only the DOM node will be removed from the body? 

You could do something like this
```js
const element1 = document.createElement('div')
const element2 = document.createElement('div')

document.body.appendChild(element1)
document.body.appendChild(element2)

riot.mount(element1, {}, 'my-component-red')
riot.mount(element2, {}, 'my-component-green')
riot.unmount(element1, true)
riot.unmount(element2)

expect(document.body.contains(element1)).to.be.ok
expect(document.body.contains(element2)).to.be.not.ok

element2.parentNode.removeChild(element2)

```

Reply:
Thanks for the suggestions.
i've fixed the test spec but there seems be a problem, it fails on the second test:

`expect(document.body.contains(element2)).to.be.not.ok`

I've tried to revert my changes and use just that test but it also fails, i'm not sure but it seems that the element is kept even without the keepRootElement argument.
I will try to check it further and will inform you latter of what i found about it.

Reply:
ok thank you. I will check what's the problem

Reply:
Ok, if is something i can do i won't mind finishing it.
I have updated the test spec, it should be ok now and it is passing the tests, but while i did it i found something that i'm not sure if is the way it is supposed to work, maybe it can helps.

In the test spec you provided, the element is registered without a template, and when that happens the previous test fails because even without the extra keepRootElement argument the root element stays in the dom at the unmount.
In an opposite way if it has a template, the root element will be removed at unmount, as i think it was supposed.

I'm not sure if it's the way it is supposed to work, but i've prepared a sample test for you to check

[https://plnkr.co/edit/uAme9WrgLW1N0ZMNrOs7](https://plnkr.co/edit/uAme9WrgLW1N0ZMNrOs7?p=preview)

Cheers